webapp/javascript/components/ComparisonApp.jsx

```diff
@@ -61,6 +106,11 @@ function ComparisonApp(props) {
               data-testid="flamegraph-renderer-left"
               display="both"
               rawFlamegraph={rawLeft}
+              isSearchLinked={linkedSearch.isSearchLinked}
+              setSearchQuery={setSearchQuery}
+              linkedSearchQuery={linkedSearch.linkedSearchQuery}
+              toggleLinkedSearch={toggleLinkedSearch}
+              resetLinkedSearchSide={linkedSearch.resetLinkedSearchSide}
```

Comment:
This is case where lifting state up makes the parent component too bloated. I think you can infer based on the toolbar `viewType` whether to show the link, and then dispatch a redux action to enable/disable the link functionality.